src/modules/common/components/Headroom.js

```diff
@@ -0,0 +1,141 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import raf from 'raf'
+
+import style from './Headroom.css'
+
+const UPWARDS = 'up'
+const DOWNWARDS = 'down'
+
+class Headroom extends React.PureComponent {
+  static propTypes = {
+    children: PropTypes.any.isRequired, // The child node to be displayed as a header
+    scrollHeight: PropTypes.number.isRequired, // The maximum amount of px the header should move up when scrolling
+    pinStart: PropTypes.number.isRequired, // The minimum scrollTop position where the transform should start
+    stickyAncestor: PropTypes.node, // Gets rendered with a corresponding stickyTop prop as an ancestor
```

Comment:
I'm ok with that formatting as I could not find a JSDoc example for documenting values of plain objects.